lib/api_remote.ml

```diff
@@ -134,22 +117,22 @@ let complete_upload_external ~(ctx : Context.t) ~(req : Slack_t.complete_upload_
       with content [file.content]; Not supporting file upload through form using
       `file` currently
       uses web API with access token *)
-let upload_file ~(ctx : Context.t) ~(file : Slack_t.files_upload_req) =
-  match file.filename, file.content with
+let upload_file ~(ctx : Context.t) ~(req : Slack_t.files_upload_req) =
+  match req.filename, req.content with
   | None, _ | _, None -> Exn.fail "need to supply both filename and content"
   | Some filename, Some content ->
     let length = String.length content in
-    let req = Slack_j.make_get_upload_url_ext_req ~filename ~length () in
-    ( match%lwt get_upload_url_external ~ctx ~req with
+    let req' = Slack_j.make_get_upload_url_ext_req ~filename ~length () in
+    ( match%lwt get_upload_url_external ~ctx ~req:req' with
     | Error e -> Lwt.return_error e
     | Ok { upload_url; file_id; _ } ->
       let raw_file_contents = In_channel.with_open_bin filename (fun ic -> input_all ic) in
```

Comment:
Please add the changes requested in https://github.com/ahrefs/slack/pull/11#discussion_r1946809020 too